src/index.ts

```diff
@@ -1,3 +1,111 @@
-export function init() {
-  return 'init...';
+import { IMetaQuerier, BasicMetaQuerier } from './util/metaquerier';
+import {
+  matchPattern,
+  shouldStartNewOne,
+  isDragStep,
+  shouldStopCurrentOne,
+  needCollect,
+} from './match-pattern';
+import { EventObserver } from './observer';
+import { Step, StepEvent } from './types';
+import { EventHandler } from './util/eventbus';
+
+type StepEventHandler = (step: Step) => void;
+
+export type RecorderOptions = {
+  doc: Document;
+  win: Window;
+  onEmit: StepEventHandler;
+  metaQuerier?: IMetaQuerier;
+  onMouseDown?: EventHandler;
```

Comment:
What's the use case of these event handler options?

Looks like when a user defines an event handler, e.g, `onMouseDown`. It will replace the built-in mouse down handler.
The customize `onMouseDown` will get a raw DOM event as the parameter but looks like it could not interact with the recorder anymore.

Reply:
I think I give it too much flexibility and I will do some change to eventbus to limit its usage.

Reply:
I think we should decoupling recorder and observer here and expose interface for recorder to communicate with incoming event info.